src/org/traccar/protocol/TelicProtocolDecoder.java

```diff
@@ -105,4 +110,57 @@ protected Object decode(
         return position;
     }
 
+    private String decodeAlarm(String event) {
+
+        int eventId = 0;
+
+        try {
+            eventId = Integer.parseInt(event);
+        } catch (Exception e) {
+            Log.error("TelicProtocolDecoder: Error parsing event (" + eventId + "): " + e.toString());
+        }
+
+        switch (eventId) {
+            case 1:
+                return Position.ALARM_POWER_ON; // Power on
+            case 2:
+                return Position.ALARM_SOS; // Emergency
+            case 3:
+                return null; // Position lock alarm
+            case 4:
+                return null; // Alarm Tracking
+            case 5:
+                return Position.ALARM_POWER_OFF; // Power off
+            case 6:
+                return null; // Course change
+            case 7:
+                return Position.ALARM_GEOFENCE_ENTER; // Geofence area enter
+            case 8:
+                return Position.ALARM_GEOFENCE_EXIT; // Geofence area exit
+            case 9:
+                return null; // GPS fix lost
+            case 10:
+                return null; // Periodic wakeup / Routine message
```

Comment:
Some of the ids are not alarms, so you shouldn't handle them. Most of the nulls seem to be just causes for the message, but not alarms. So we don't need cases for those ids here.